getting-started/bookshelf/config.go

```diff
@@ -12,133 +12,92 @@
-	// localhost:3306 is used, and the instance name is ignored.
-	// DB, err = configureCloudSQL(cloudSQLConfig{
-	// 	Username: "root",
-	// 	Password: "",
-	// 	// The connection name of the Cloud SQL v2 instance, i.e.,
-	// 	// "project:region:instance-id"
-	// 	// Cloud SQL v1 instances are not supported.
-	// 	Instance: "",
-	// })
-	// [END cloudsql]
-
-	// [START mongo]
-	// To use Mongo, uncomment the next lines and update the address string and
-	// optionally, the credentials.
-	//
-	// var cred *mgo.Credential
-	// DB, err = newMongoDB("localhost", cred)
-	// [END mongo]
-
-	// [START datastore]
-	// To use Cloud Datastore, uncomment the following lines and update the
-	// project ID.
-	// More options can be set, see the google package docs for details:
-	// http://godoc.org/golang.org/x/oauth2/google
-	//
-	// DB, err = configureDatastoreDB("<your-project-id>")
-	// [END datastore]
+// Bookshelf holds a BookDatabase and storage info.
+type Bookshelf struct {
+	DB BookDatabase
 
-	if err != nil {
-		log.Fatal(err)
-	}
+	StorageBucket     *storage.BucketHandle
+	StorageBucketName string
 
-	// [START storage]
-	// To configure Cloud Storage, uncomment the following lines and update the
-	// bucket name.
-	//
-	// StorageBucketName = "<your-storage-bucket>"
-	// StorageBucket, err = configureStorage(StorageBucketName)
-	// [END storage]
+	// logWriter is used for request logging and can be overridden for tests.
+	logWriter io.Writer
+}
 
-	if err != nil {
-		log.Fatal(err)
+// NewBookshelf creates a new Bookshelf.
+func NewBookshelf(projectID string) (*Bookshelf, error) {
+	b := &Bookshelf{
+		logWriter: os.Stderr,
 	}
 
-	// [START auth]
-	// To enable user sign-in, uncomment the following lines and update the
-	// Client ID and Client Secret.
-	// You will also need to update OAUTH2_CALLBACK in app.yaml when pushing to
-	// production.
-	//
-	// OAuthConfig = configureOAuthClient("clientid", "clientsecret")
-	// [END auth]
-
-	// [START sessions]
-	// Configure storage method for session-wide information.
-	// Update "something-very-secret" with a hard to guess string or byte sequence.
-	cookieStore := sessions.NewCookieStore([]byte("something-very-secret"))
-	cookieStore.Options = &sessions.Options{
-		HttpOnly: true,
+	var err error
+	b.DB, err = configureFirestoreDB(projectID)
+	if err != nil {
+		return nil, err
 	}
-	SessionStore = cookieStore
-	// [END sessions]
-
-	// [START pubsub]
-	// To configure Pub/Sub, uncomment the following lines and update the project ID.
-	//
-	// PubsubClient, err = configurePubsub("<your-project-id>")
-	// [END pubsub]
 
+	// This Cloud Storage bucket must exist to be able to upload book pictures.
+	// You can create it and make it public by running:
+	//     gsutil mb my-project_bucket
+	//     gsutil defacl set public-read gs://my-project_bucket
+	// replacing my-project with your project ID.
+	b.StorageBucketName = projectID + "_bucket"
```

Comment:
Why the `_bucket` suffix?

Since this is going to GAE, is the `<project>.appspot.com` "default bucket" not suitable? I can't remember whether this is meant to be used or not.

Reply:
If the bucket wasn't created, I think the first error the user/developer will see is when they upload a picture. That seems OK but maybe a friendly message to tell them the commands to make the bucket would be good.

Reply:
Matching the existing tutorial with the name.

The error when you try to upload an image is "could not parse book from form: could not upload file: bucket "my-project_bucket" does not exist: check config.go". The `gsutil mb` and `defacl` explanation is a bit long, so I left it as a reference to config.go.